scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -3032,26 +3031,24 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
   }
 
   def blockExpr(): Term = autoPos {
-    def blockOrCase() = {
-      if (token.is[CaseIntro] || (token.is[Ellipsis] && ahead(token.is[KwCase])))
-        Term.PartialFunction(caseClauses())
-      else block()
-    }
-    if (token.is[LeftBrace]) {
-      inBraces(blockOrCase())
-    } else if (token.is[Indentation.Indent]) {
-      dropTrivialBlock(indented(blockOrCase()))
-    } else {
-      syntaxError("Expected brace or indentation.", at = token.pos)
-    }
+    if (ahead(token.is[CaseIntro] || (token.is[Ellipsis] && ahead(token.is[KwCase])))) {
+      if (token.is[LeftBrace])
+        Term.PartialFunction(inBraces(caseClauses()))
+      else
+        Term.PartialFunction(indented(caseClauses()))
+    } else block()
   }
 
   def block(): Term = autoPos {
-    if (token.is[Indentation.Indent])
-      indented(
-        dropTrivialBlock(Term.Block(blockStatSeq()))
-      )
-    else Term.Block(blockStatSeq())
+    if (token.is[LeftBrace]) {
+      Term.Block(inBraces(blockStatSeq()))
+    } else {
+      val possibleBlock =
+        if (token.is[Indentation.Indent])
+          indented(Term.Block(blockStatSeq()))
```

Comment:
above, you did inBraces and then Block. here, first Block and then indented. does the order matter?

if it doesn't, I'd first create Block over blockStatSeq as a variable, then overlay inBraces or indented.

Reply:
The order doesn't matter in this case, changed!